lightning/mydump/csv_parser.go

```diff
@@ -66,18 +68,22 @@ func NewCSVParser(
 	ioWorkers *worker.Pool,
 	shouldParseHeader bool,
 ) *CSVParser {
-	quote := byte(0)
+	quote := []byte{0}
 	if len(cfg.Delimiter) > 0 {
-		quote = cfg.Delimiter[0]
+		quote = []byte(cfg.Delimiter)
 	}
```

Comment:
with multi-char support, the "no quote" value should now be the empty string instead of "\0".

```suggestion
	quote := []byte(cfg.Delimiter)
```